lib/ukofw/Makefile.uk

```diff
@@ -5,4 +5,3 @@ CINCLUDES-$(CONFIG_LIBUKOFW)   += -I$(LIBUKOFW_BASE)/include
 CXXINCLUDES-$(CONFIG_LIBUKOFW) += -I$(LIBUKOFW_BASE)/include
 
 LIBUKOFW_SRCS-y                += $(LIBUKOFW_BASE)/fdt.c
-LIBUKOFW_SRCS-y                += $(LIBUKOFW_BASE)/gic_fdt.c
```

Comment:
> Maybe in the future with libfwif we could just have a generic firmware node ops method equivalent to that of of_xlate from Linux/U-Boot

I don't follow, can you please elaborate? Actually I thought for a bit whether there should be a centralized place for such helpers or whether each driver should provide its own when they are not resuable, and concluded on the latter. Notice that the interpretation of the `interrupts` property is interrupt controller specific, ie for GIC it is defined in https://www.kernel.org/doc/Documentation/devicetree/bindings/interrupt-controller/arm%2Cgic.yaml, which means that in theory another architecture that uses fdt may provide a different interpretation using a different number of `interrrupt-cells` and so on.

Reply:
Yes, I understood. For such cases U-Boot, for example, allows the registration of an `of_xlate` method that does this device specific translation, e.g. for [clocks](https://github.com/u-boot/u-boot/blob/master/include/clk-uclass.h#L32), as depending on the clock in the clock-tree it may interpret the size/cells differently.

Reply:
Actually what is `libfwif`? 🤔 I don't remember discussing about it before. Either way, what I understand from the above is the proposal to move the fdt parsing from a function in the ukintctlr API into driver ops because a similar pattern can be established for other types of devices. Is that correct?

Nevertheless, the rest of the GIC ops are interrupt-controller specific, and from what I see similarly for u-boot they add this function to other device-specific ops, at least for clocks and irq that I checked on. So it doesn't seem they follow a generic device API that includes such a function. I would had expected such a thing instead, ie a generic device API with common ops for all types of devices, along with a set of device-specific ops. But all this is out of scope of this PR.

Btw, there was actually a translate function in gic ops that was only taking care of the translation of the interrupt number, and I got rid of it in favor of a more fully-fledged parsing function of the `interrupts` property, perhaps you noticed that already.

Reply:
Yes, ok, I was probably not clear enough, I am not asking for any change. I am simply mentioning that in the future it would be good to have a generic `of_xlate`-like method that a driver could implement with the help of a `libukfwif` that would abstract away access to DT/AML, but for now all fine. :+1: I am not requesting any changes.